### ISSUE TITLE
Running on Ubuntu 16

### ISSUE THREAD

**Issue by raducrisan1** [Microsoft/vscode-cordova#156]
Hello,
I have two issues using the extension from Ubuntu 16.04. I tried on different machines and the behavior is the same all the time. 

I wrote down the steps I followed to set up the Cordova app:

I have a fresh install of Ubuntu 16.04, which has node 4.2.6
I installed cordova globally from bash: `sudo npm install -g cordova` . The installed version is 6.2.0
I created a first app from bash: `cordova create simpleapp`
From simpleapp directory I added the android platform: `cordova platform add android`. The installed version is 5.1.1
I installed Android Studio 2.1.2 to help me have the emulator and the Android SDK installed. Here is the path I added into the ~/.bashrc file: `PATH=$PATH:/home/radu/Android/Sdk/platform-tools:/home/radu/Android/Sdk/tools`. I also added the ANDROID_HOME environment variable. The output of `echo $ANDROID_HOME` is: `/home/radu/Android/Sdk`
I created a single Android Virtual Device based on Marshmallow X64 system image. Then I opened the simpleapp folder from VS Code

The first issue is related to build. I pressed F6 to build the app and the message coming from the extension is: `/bin/sh: 1: cordova.cmd: not found`, although I am on Ubuntu 16.04.
![buildcordova](https://cloud.githubusercontent.com/assets/15193363/16430110/a0bc50c8-3d82-11e6-91c5-7120b54ac005.png)

The second issue is related to debugging: I couldn't properly use the debug mode, VS Code did not attach to the emulator. In the debug mode I selected Run Android on Emulator and then I pressed F5 to start debugging. The emulator started, the app opened and after the message `Attaching to android` I encountered another message: `Unable to find emulator`. 
![debugcordova](https://cloud.githubusercontent.com/assets/15193363/16430252/6558371c-3d83-11e6-943c-5c1520866eb6.png)

The emulator had the simple app in front, like below:
![emulator](https://cloud.githubusercontent.com/assets/15193363/16430339/d6ae034c-3d83-11e6-9853-e8811dcf2384.png)

Please let me know what to do in order to have the issues fixed. Thank you!

**Comment by MSLaguana** [comment]
Thanks for reporting these issues. We haven't done a lot of testing outside of windows and mac, and it looks like we've got some platform-dependent bits in there that don't account for linux.

For your first issue, we have some platform checking code backwards (just put a PR out for fixing it #157), but you can put in a workaround by going to the install location of the vscode extension (I believe it will be something like `~/.vscode/extensions/vsmobile.cordova-tools-1.2.0` and then opening out/src/utils/cordovaCommandHelper.js` and replacing

```
    CordovaCommandHelper.CORDOVA_CMD_NAME = os.platform() === "darwin" ? "cordova" : "cordova.cmd";
    CordovaCommandHelper.IONIC_CMD_NAME = os.platform() === "darwin" ? "ionic" : "ionic.cmd";
```
with
```
    CordovaCommandHelper.CORDOVA_CMD_NAME = os.platform() === "win32" ? "cordova.cmd" : "cordova";
    CordovaCommandHelper.IONIC_CMD_NAME = os.platform() === "win32" ? "ionic.cmd" : "ionic";
```

For the second issue we'll need to troubleshoot further. When you have the emulator running as you show above, what is the output of `adb devices`?

**Comment by raducrisan1** [comment]
Hi,
The output is:
```
List of devices attached
emulator-5554	device
```
Many thanks for the rocket fast response.

**Comment by MSLaguana** [comment]
We are trying to match that against the regex here: https://github.com/Microsoft/vscode-cordova/blob/master/src/debugger/cordovaDebugAdapter.ts#L209

Visually that looks ok to me, but would you mind running `adb devices > out.tmp` and uploading that file somewhere? We assume/require that the gap between emulator-5554 and device is a tab, and it's possible that there is some whitespace difference we don't account for.

**Comment by raducrisan1** [comment]
[out.tmp.gz](https://github.com/Microsoft/vscode-cordova/files/338043/out.tmp.gz)

Done.

**Comment by MSLaguana** [comment]
Actually this sounds similar to another issue also reported on ubuntu: https://github.com/Microsoft/vscode-cordova/issues/100

I haven't had the chance to investigate on a linux machine capable of running an android emulator yet, but for a very unusual workaround, you could try running vscode under strace...

**Comment by MSLaguana** [comment]
That file looks perfectly find and behaves fine when I try to interact with it on windows... It may be something specific to when we run the command in our process, but i'm not sure when I'll have a chance to set up a ubuntu machine to test things out on I'm afraid.

**Comment by raducrisan1** [comment]
There's a strange thing here. I followed your advice but with a small change, I actually ran `strace -o /dev/null -f code .` and it successfully attached to the emulator :)

Thanks for your help.

**Comment by MSLaguana** [comment]
I'm glad we could get things working for you! I really need to find out what's actually going on here, because it's bizzare.

**Comment by dlebu** [comment]
@raducrisan1 Thanks again for reporting this issue and for using our tools. It was fixed in #158 and will ship soon in the next extension update.

**Comment by dlebu** [comment]
@raducrisan1 The updated version of the extension is published. You can update by typing `ext update` in your command palette. Please keep the feedback coming!
